plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/JdbcIcebergClient.java

```diff
@@ -0,0 +1,246 @@
+        }
+    }
+
+    public Optional<String> getNamespaceLocation(String namespace)
+    {
+        try (Handle handle = Jdbi.open(connectionFactory)) {
+            return handle.createQuery("" +
+                            "SELECT value " +
+                            "FROM iceberg_namespace_properties " +
+                            "WHERE catalog_name = :catalog AND namespace = :schema AND key = 'location'")
+                    .bind("catalog", catalogId)
+                    .bind("schema", namespace)
+                    .mapTo(String.class)
+                    .findOne();
+        }
+    }
+
+    public void createNamespace(String namespace, Map<String, Object> properties)
+    {
+        ImmutableList.Builder<NamespaceProperties> namespaceProperties = ImmutableList.builderWithExpectedSize(properties.size() + 1);
+        namespaceProperties.add(new NamespaceProperties(catalogId, namespace, "exists", "true"));
+        for (Entry<String, Object> property : properties.entrySet()) {
+            namespaceProperties.add(new NamespaceProperties(catalogId, namespace, property.getKey(), property.getValue().toString()));
+        }
+
+        try (Handle handle = Jdbi.open(connectionFactory)) {
+            handle.createUpdate("" +
+                            "INSERT INTO iceberg_namespace_properties " +
+                            "(catalog_name, namespace, key, value) " +
+                            "VALUES <values>")
+                    .bindBeanList("values", namespaceProperties.build(), ImmutableList.of("catalogName", "namespace", "key", "value"))
+                    .execute();
+        }
+    }
+
+    public void dropNamespace(String namespace)
+    {
+        try (Handle handle = Jdbi.open(connectionFactory)) {
+            handle.createUpdate("" +
+                            "DELETE FROM iceberg_namespace_properties " +
+                            "WHERE catalog_name = :catalog AND namespace = :schema")
+                    .bind("catalog", catalogId)
+                    .bind("schema", namespace)
+                    .execute();
+        }
+    }
+
+    public List<SchemaTableName> getTables(List<String> namespaces)
+    {
+        try (Handle handle = Jdbi.open(connectionFactory)) {
+            return handle.createQuery("" +
+                    "SELECT table_namespace, table_name " +
+                    "FROM iceberg_tables " +
+                    "WHERE catalog_name = :catalog AND table_namespace IN (<namespaces>)")
+                    .bind("catalog", catalogId)
+                    .bindList("namespaces", namespaces)
+                    .map((rs, ctx) -> new SchemaTableName(rs.getString("table_namespace"), rs.getString("table_name")))
+                    .list();
+        }
+    }
+
+    public void createTable(String schemaName, String tableName, String metadataLocation)
+    {
+        try (Handle handle = Jdbi.open(connectionFactory)) {
+            handle.createUpdate("" +
```

Comment:
can `org.apache.iceberg.jdbc.JdbcCatalog:: newCreateTableTransaction` be used directly?